### PR TITLE
Change/mining reservers

### PR DESCRIPTION
**Pull request by karlthepagan**
this requires #655 

an experimental change which de-prioritizes reservers until a remote mine's haulers have been built

### CODE REVIEW

**Comment by karlthepagan** [comment]
Related to #621

**Comment by karlthepagan** [comment]
after observing this casually I'm decreasing the weight requirement to 20%

**Comment by karlthepagan** [comment]
rebased onto #655

**Comment by pandabear41** [comment]
Line 237 and 243. The logic changed for reservation checking and current code was causing reservers to attempt to spawn for SK rooms. 
e72cb34d8462ee5a0baf32129a7c1ed92438d999

**Comment by karlthepagan** [comment]
> but I don't believe people want to take into account queued creeps when considering capacity, only running or spawning creeps.

That's what happens here, but I guess it's not super clear. The only time queued creeps are considered is when calculating creeps to put into the queue.

**Comment by karlthepagan** [comment]
The final requirement is 10% of capacity. This basically means that a miner and hauler will spawn first before the reserver, but once that is setup the reserver's medium priority takes over.

File: task.mining.js [hunk 1]
```diff
@@ -421,15 +421,30 @@ function haulerCarryToWeight(carry) {
...
+
+            if( population < REMOTE_RESERVE_HAUL_CAPACITY ) {
+                // TODO if this room & all exits are currently reserved (by anyone) then use default to prevent Invaders?
+                if( DEBUG && TRACE ) trace('Task', {flagName:flag.name, pos:flag.pos, population, spawnParams:'population', [mod.name]:'spawnParams', Task:mod.name});
+                return function() { // TODO how to pass thru partial results?
+                    return {count: 0, priority: 'Low'};
+                };
+            }
+
+            // undefined means use defaults
```
Comment (karlthepagan) [inline task.mining.js]: This is non-standard for strategies so I'm having a real good think about it before I merge into dev.

**Pull request merged**